Mathlib/Order/Category/BddOrd.lean

```diff
@@ -19,74 +19,155 @@ universe u v
 
 /-- The category of bounded orders with monotone functions. -/
-structure BddOrd where
-  /-- The underlying object in the category of partial orders. -/
-  toPartOrd : PartOrd
+structure BddOrd extends PartOrd where
   [isBoundedOrder : BoundedOrder toPartOrd]
 
+/-- The underlying object in the category of partial orders. -/
+add_decl_doc BddOrd.toPartOrd
+
+attribute [instance] BddOrd.isBoundedOrder
+
+initialize_simps_projections BddOrd (carrier → coe, -str)
+
 namespace BddOrd
 
 instance : CoeSort BddOrd Type* :=
   InducedCategory.hasCoeToSort toPartOrd
 
-instance (X : BddOrd) : PartialOrder X :=
-  X.toPartOrd.str
+/-- Construct a bundled `BddOrd` from the underlying type and typeclass. -/
+abbrev of (X : Type*) [PartialOrder X] [BoundedOrder X] : BddOrd where
+  carrier := X
 
-attribute [instance] BddOrd.isBoundedOrder
+/-- The type of morphisms in `BddOrd R`. -/
+@[ext]
+structure Hom (X Y : BddOrd.{u}) where
+  private mk ::
+  /-- The underlying `BoundedOrderHom`. -/
+  hom' : BoundedOrderHom X Y
+
+instance : Category BddOrd.{u} where
+  Hom X Y := Hom X Y
+  id _ := ⟨BoundedOrderHom.id _⟩
+  comp f g := ⟨g.hom'.comp f.hom'⟩
+
+instance : ConcreteCategory BddOrd (BoundedOrderHom · ·) where
+  hom := Hom.hom'
+  ofHom := Hom.mk
+
+/-- Turn a morphism in `BddOrd` back into a `BoundedOrderHom`. -/
+abbrev Hom.hom {X Y : BddOrd.{u}} (f : Hom X Y) :=
+  ConcreteCategory.hom (C := BddOrd) f
+
+/-- Typecheck a `BoundedOrderHom` as a morphism in `BddOrd`. -/
+abbrev ofHom {X Y : Type u} [PartialOrder X] [BoundedOrder X] [PartialOrder Y] [BoundedOrder Y]
+    (f : BoundedOrderHom X Y) : of X ⟶ of Y :=
+  ConcreteCategory.ofHom (C := BddOrd) f
+
+variable {R} in
+/-- Use the `ConcreteCategory.hom` projection for `@[simps]` lemmas. -/
+def Hom.Simps.hom (X Y : BddOrd.{u}) (f : Hom X Y) :=
+  f.hom
+
+initialize_simps_projections Hom (hom' → hom)
+
+/-!
+The results below duplicate the `ConcreteCategory` simp lemmas, but we can keep them for `dsimp`.
+-/
+
+@[simp]
+lemma coe_id {X : BddOrd} : (𝟙 X : X → X) = id := rfl
+
+@[simp]
+lemma coe_comp {X Y Z : BddOrd} {f : X ⟶ Y} {g : Y ⟶ Z} : (f ≫ g : X → Z) = g ∘ f := rfl
+
+@[simp]
+lemma forget_map {X Y : BddOrd} (f : X ⟶ Y) :
+    (forget BddOrd).map f = f := rfl
+
+@[ext]
+lemma ext {X Y : BddOrd} {f g : X ⟶ Y} (w : ∀ x : X, f x = g x) : f = g :=
+  ConcreteCategory.hom_ext _ _ w
 
-/-- Construct a bundled `BddOrd` from a `Fintype` `PartialOrder`. -/
-def of (α : Type*) [PartialOrder α] [BoundedOrder α] : BddOrd :=
-  -- Porting note: was ⟨⟨α⟩⟩, see https://github.com/leanprover-community/mathlib4/issues/4998
-  ⟨{ α := α }⟩
+-- This is not `simp` to avoid rewriting in types of terms.
+theorem coe_of (X : Type u) [PartialOrder X] [BoundedOrder X] : (BddOrd.of X : Type u) = X := rfl
 
 @[simp]
-theorem coe_of (α : Type*) [PartialOrder α] [BoundedOrder α] : ↥(of α) = α :=
+lemma hom_id {X : BddOrd} : (𝟙 X : X ⟶ X).hom = BoundedOrderHom.id _ := rfl
+
+/- Provided for rewriting. -/
```

Comment:
Is this meant to be a docstring?

Reply:
It's more of a comment explaining why this isn't marked as `@[simp]`. But we can docstringify all of these comments.

Reply:
I'm just surprised you're using single-line multi-line comments